nocgo.go

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build !cgo && !ios
+// +build !cgo,!ios
```

Comment:
Isn't `!cgo` enough? It's because `!cgo` & `ios` is an impossible combination so far.

Reply:
This way, running `CGO_ENABLED=0 GOOS=ios go build ./cmd/main.go`
```
# command-line-arguments
/Users/jarrettkuklis/go/go1.19/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
Undefined symbols for architecture arm64:
  "__cgo_yield", referenced from:
      _runtime.cgo_yield in go.o
  "_x_cgo_init", referenced from:
      __cgo_init in go.o
  "_x_cgo_notify_runtime_init_done", referenced from:
      __cgo_notify_runtime_init_done in go.o
  "_x_cgo_setenv", referenced from:
      _runtime._cgo_setenv in go.o
  "_x_cgo_thread_start", referenced from:
      __cgo_thread_start in go.o
  "_x_cgo_unsetenv", referenced from:
      _runtime._cgo_unsetenv in go.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
If `ios` is removed from `nocgo.go` then you get this:
`2022/08/19 09:02:27 duplicated definition of symbol _cgo_init, from github.com/ebitengine/purego/internal/fakecgo and runtime/cgo`.

Which error is more helpful?

Reply:
I think both are not so helpful and don't have significant change. Then, I slighly prefer not having ios. What do you think?

Reply:
I think removing ios is better bc it at least mentions fakecgo so it’d be easier to spot and suggest a fix (ie Enable Cgo)

Reply:
It is possible to show a pseudo compile error message like this

```go
//go:build !cgo && ios
// +build !cgo,ios

var _ = _IOS_REQUIRES_CGO
```

This is optional. I'm fine with both having and not having this.

Reply:
I like it! I also added a comment that describes what the error means.